File: rust/src/airport.rs
use std::collections::HashMap;

/// Returns a map from "city,state" (lowercase) to IATA airport code.
fn airports() -> HashMap<&'static str, &'static str> {
    let mut m = HashMap::new();
    m.insert("new york,ny",      "JFK");
    m.insert("los angeles,ca",   "LAX");
    m.insert("chicago,il",       "ORD");
    m.insert("houston,tx",       "IAH");
    m.insert("phoenix,az",       "PHX");
    m.insert("philadelphia,pa",  "PHL");
    m.insert("san antonio,tx",   "SAT");
    m.insert("san diego,ca",     "SAN");
    m.insert("dallas,tx",        "DFW");
    m.insert("san jose,ca",      "SJC");
    m.insert("austin,tx",        "AUS");
    m.insert("jacksonville,fl",  "JAX");
    m.insert("san francisco,ca", "SFO");
    m.insert("columbus,oh",      "CMH");
    m.insert("charlotte,nc",     "CLT");
    m.insert("indianapolis,in",  "IND");
    m.insert("seattle,wa",       "SEA");
    m.insert("denver,co",        "DEN");
    m.insert("nashville,tn",     "BNA");
    m.insert("oklahoma city,ok", "OKC");
    m.insert("el paso,tx",       "ELP");
    m.insert("washington,dc",    "DCA");
    m.insert("las vegas,nv",     "LAS");
    m.insert("louisville,ky",    "SDF");
    m.insert("baltimore,md",     "BWI");
    m.insert("milwaukee,wi",     "MKE");
    m.insert("albuquerque,nm",   "ABQ");
    m.insert("tucson,az",        "TUS");
    m.insert("fresno,ca",        "FAT");
    m.insert("sacramento,ca",    "SMF");
    m.insert("kansas city,mo",   "MCI");
    m.insert("atlanta,ga",       "ATL");
    m.insert("miami,fl",         "MIA");
    m.insert("minneapolis,mn",   "MSP");
    m.insert("portland,or",      "PDX");
    m.insert("detroit,mi",       "DTW");
    m.insert("boston,ma",        "BOS");
    m.insert("memphis,tn",       "MEM");
    m.insert("new orleans,la",   "MSY");
    m.insert("cleveland,oh",     "CLE");
    m.insert("tampa,fl",         "TPA");
    m.insert("orlando,fl",       "MCO");
    m
}

/// Returns the IATA airport code for a given city and state.
///
/// # Errors
/// Returns an error string if no airport is found for the city/state pair.
pub fn lookup_airport(city: &str, state: &str) -> Result<&'static str, String> {
    let key = format!("{},{}", city.trim().to_lowercase(), state.trim().to_lowercase());
    airports()
        .get(key.as_str())
        .copied()
        .ok_or_else(|| format!("no airport found for {}, {}", city, state))
}

File: src/airport.ts
const airports: Record<string, string> = {
  "new york,ny": "JFK",
  "los angeles,ca": "LAX",
  "chicago,il": "ORD",
  "houston,tx": "IAH",
  "phoenix,az": "PHX",
  "philadelphia,pa": "PHL",
  "san antonio,tx": "SAT",
  "san diego,ca": "SAN",
  "dallas,tx": "DFW",
  "san jose,ca": "SJC",
  "austin,tx": "AUS",
  "jacksonville,fl": "JAX",
  "san francisco,ca": "SFO",
  "columbus,oh": "CMH",
  "charlotte,nc": "CLT",
  "indianapolis,in": "IND",
  "seattle,wa": "SEA",
  "denver,co": "DEN",
  "nashville,tn": "BNA",
  "oklahoma city,ok": "OKC",
  "el paso,tx": "ELP",
  "washington,dc": "DCA",
  "las vegas,nv": "LAS",
  "louisville,ky": "SDF",
  "baltimore,md": "BWI",
  "milwaukee,wi": "MKE",
  "albuquerque,nm": "ABQ",
  "tucson,az": "TUS",
  "fresno,ca": "FAT",
  "sacramento,ca": "SMF",
  "kansas city,mo": "MCI",
  "atlanta,ga": "ATL",
  "miami,fl": "MIA",
  "minneapolis,mn": "MSP",
  "portland,or": "PDX",
  "detroit,mi": "DTW",
  "boston,ma": "BOS",
  "memphis,tn": "MEM",
  "new orleans,la": "MSY",
  "cleveland,oh": "CLE",
  "tampa,fl": "TPA",
  "orlando,fl": "MCO",
};

export function lookupAirport(city: string, state: string): string {
  const key = `${city.trim().toLowerCase()},${state.trim().toLowerCase()}`;
  const code = airports[key];
  if (!code) {
    throw new Error(`No airport found for ${city}, ${state}`);
  }
  return code;
}
